NotificationHubs/src/com/windowsazure/messaging/RetryUtil.java

```diff
@@ -72,18 +72,46 @@ public static RetryPolicy getRetryPolicy(RetryOptions options) {
                 }
 
                 if (error instanceof TimeoutException) {
-                	System.out.println("TimeoutException error occurred. Retrying operation. Attempt: " + attempt);
-                	
+                	System.out.println("TimeoutException error occurred. Retrying operation. Attempt: " + attempt); 
                     return retryPolicy.calculateRetryDelay(error, attempt);
-                } else if (error instanceof QuotaExceededException && (((QuotaExceededException) error).getIsTransient())) {
+                } else if (error instanceof QuotaExceededException) {
                 	System.out.println("Retryable error occurred. Retrying operation. Attempt: " + attempt + ". Error: " + error);
-                	
                     return retryPolicy.calculateRetryDelay(error, attempt);
+                } else if (error instanceof NotificationHubsException) {                    	
+                    	int statusCode = ((NotificationHubsException) error).getHttpStatusCode();
+                    	if (statusCode == 408 || statusCode == 500 || statusCode == 503 || statusCode == 504) {
```

Comment:
looks like we should be using the `retriableStatusCodes` member variable here like: `retriableStatusCodes.contains(statusCode)`;